components/applications-service/pkg/storage/postgres/hab_event.go

```diff
@@ -76,64 +99,113 @@ func (db *Postgres) IngestHabEventWithoutMetrics(event *applications.HabService)
+		if svcMetadata.GetUpdateConfig() != nil {
+			svc.Channel = svcMetadata.UpdateConfig.GetChannel()
+		}
+
 		if _, err := db.DbMap.Update(svc); err != nil {
-			return errors.Wrap(err, "Unable to insert service")
+			return errors.Wrap(err, "unable to update service")
 		}
 		return nil
 	}
 
 	// But if the service doesn't exist, we will handle it as a new service insertion
-	return db.insertNewService(event)
+	return db.insertNewServiceFromHealthCheckEvent(
+		eventMetadata,
+		svcMetadata,
+		pkgIdent,
+		strings.ToUpper(event.GetResult().String()),
+	)
 }
 
-// insertNewService assumes the service to be inserted doesn't exist already,
+// insertNewServiceFromHealthCheckEvent assumes the service to be inserted doesn't exist already,
 // this function is wrapped with a transaction func since it could do multiple things
 // like insert a supervisor, service_group, deployment and the service itself.
-func (db *Postgres) insertNewService(event *applications.HabService) error {
+func (db *Postgres) insertNewServiceFromHealthCheckEvent(
+	eventMetadata *habitat.EventMetadata,
+	svcMetadata *habitat.ServiceMetadata,
+	pkgIdent *packageIdent,
+	health string) error {
 
+	// TODO Update me!
 	return dblib.Transaction(db.DbMap, func(tx *gorp.Transaction) error {
 
 		// 1) Deployment
-		did, exist := db.getDeploymentID(event.GetApplication(), event.GetEnvironment())
+		did, exist := db.getDeploymentID(eventMetadata.GetApplication(), eventMetadata.GetEnvironment())
 		if !exist {
-			deploy := deploymentFromHabEvent(event)
+			deploy := &deployment{
+				AppName:     eventMetadata.GetApplication(),
+				Environment: eventMetadata.GetEnvironment(),
+			}
 			if err := tx.Insert(deploy); err != nil {
 				return errors.Wrap(err, "Unable to insert deployment")
 			}
 			did = deploy.ID
 		}
 
 		// 2) Service Group
-		gid, exist := db.getServiceGroupID(event.FullServiceGroupName())
+		gid, exist := db.getServiceGroupID(svcMetadata.GetServiceGroup())
 		if !exist {
-			svcGroup := serviceGroupFromHabEvent(event, did)
+			svcGroup := &serviceGroup{
+				Name:         svcMetadata.GetServiceGroup(),
+				DeploymentID: did,
+			}
 			if err := tx.Insert(svcGroup); err != nil {
 				return errors.Wrap(err, "Unable to insert service_group")
 			}
 			gid = svcGroup.ID
 		}
 
 		// 3) Supervisor
-		sid, exist := db.getSupervisorID(event.GetSupervisorId())
+		sid, exist := db.getSupervisorID(eventMetadata.GetSupervisorId())
 		if !exist {
-			sup := supervisorFromHabEvent(event)
+			sup := &supervisor{
+				MemberID: eventMetadata.GetSupervisorId(),
+				Fqdn:     eventMetadata.GetFqdn(),
+				Site:     eventMetadata.GetSite(),
+			}
+
 			if err := tx.Insert(sup); err != nil {
 				return errors.Wrap(err, "Unable to insert supervisor")
 			}
 			sid = sup.ID
 		}
 
 		// 4) Service
-		svc := serviceFromHabEvent(event, did, sid, gid)
+		svc := newService(pkgIdent, health, did, sid, gid)
+		if svcMetadata.GetUpdateConfig() != nil {
```

Comment:
I think we can move this to newService, for better encapsulation. Also we should add update strategy, as channel is irrelevant if update strategy is nil we should update the business logic when retrieving this as well.